### ISSUE TITLE
this plugin doesn't work with logstash 5.x

### ISSUE THREAD

**Issue by jsvd** [logstash-plugins/logstash-output-exec#10]
```
[2016-11-02T12:55:06,006][FATAL][logstash.runner          ] An unexpected error occurred! {:error=>#<NoMethodError: undefined method `terminal' for #<LogStash::Logging::Logger:0x4b720e33>>, :backtrace=>["/tmp/logstash-5.0.0/vendor/bundle/jruby/1.9/gems/logstash-output-exec-3.0.3/lib/logstash/outputs/exec.rb:58:in `receive'", "org/jruby/RubyIO.java:4475:in `popen3'", "/private/tmp/logstash-5.0.0/vendor/jruby/lib/ruby/1.9/open3.rb:74:in `popen3'", "/tmp/logstash-5.0.0/vendor/bundle/jruby/1.9/gems/logstash-output-exec-3.0.3/lib/logstash/outputs/exec.rb:51:in `receive'", "/tmp/logstash-5.0.0/logstash-core/lib/logstash/outputs/base.rb:92:in `multi_receive'", "org/jruby/RubyArray.java:1613:in `each'", "/tmp/logstash-5.0.0/logstash-core/lib/logstash/outputs/base.rb:92:in `multi_receive'", "/tmp/logstash-5.0.0/logstash-core/lib/logstash/output_delegator_strategies/legacy.rb:19:in `multi_receive'", "/tmp/logstash-5.0.0/logstash-core/lib/logstash/output_delegator.rb:42:in `multi_receive'", "/tmp/logstash-5.0.0/logstash-core/lib/logstash/pipeline.rb:297:in `output_batch'", "org/jruby/RubyHash.java:1342:in `each'", "/tmp/logstash-5.0.0/logstash-core/lib/logstash/pipeline.rb:296:in `output_batch'", "/tmp/logstash-5.0.0/logstash-core/lib/logstash/pipeline.rb:252:in `worker_loop'", "/tmp/logstash-5.0.0/logstash-core/lib/logstash/pipeline.rb:225:in `start_workers'"]}
```

**Comment by bmarwell** [comment]
Same for me:

```
[2016-11-30T11:44:18,861][FATAL][logstash.runner          ] An unexpected error occurred! {:error=>#<NoMethodError: undefined method `terminal' for #<LogSta
sh::Logging::Logger:0x717020aa>>, :backtrace=>["/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-exec-3.0.3/lib/logstash/outputs/exec.rb:58:
in `receive'", "org/jruby/RubyIO.java:4475:in `popen3'", "/usr/share/logstash/vendor/jruby/lib/ruby/1.9/open3.rb:74:in `popen3'", "/usr/share/logstash/vendo
r/bundle/jruby/1.9/gems/logstash-output-exec-3.0.3/lib/logstash/outputs/exec.rb:51:in `receive'", "/usr/share/logstash/logstash-core/lib/logstash/outputs/ba
se.rb:92:in `multi_receive'", "org/jruby/RubyArray.java:1613:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/outputs/base.rb:92:in `multi_receiv
e'", "/usr/share/logstash/logstash-core/lib/logstash/output_delegator_strategies/legacy.rb:19:in `multi_receive'", "/usr/share/logstash/logstash-core/lib/lo
gstash/output_delegator.rb:42:in `multi_receive'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:297:in `output_batch'", "org/jruby/RubyHash.j
ava:1342:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:296:in `output_batch'", "/usr/share/logstash/logstash-core/lib/logstash/pip
eline.rb:252:in `worker_loop'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:225:in `start_workers'"]}
```

I'm using ubuntu 16.04:
```
# apt show logstash
Package: logstash
Version: 1:5.0.1-1
Status: install ok installed
Priority: extra
Section: default
Maintainer: <vagrant@packer-virtualbox-iso-1478119172>
Installed-Size: 199 MB
Homepage: http://www.elasticsearch.org/overview/logstash/
License: Apache 2.0
Vendor: Elasticsearch
Download-Size: unknown
APT-Manual-Installed: yes
APT-Sources: /var/lib/dpkg/status
Description: An extensible logging pipeline
```

I also updated all plugins, with no luck.

**Comment by markcberman** [comment]
Suyograo:

Still encountering the same error as described in this thread when I try to invoke a simple python using the logstash-output-exec plugin using logstash 5.0.0 on Mac OS X version 10.11.6.  I installed the logstash-output-exec on 12/6/2016 using the "./logstash-plugin install logstash-output-exec" command.

Here is the exec command.  (Calls a simple python program "HelloWorld.py that outputs "Hello World".

exec{
    command => 'python /Users/markberman/TextAnalytics/PythonCode/HelloWorld.py'
  }

Here is the HelloWorld.py script (I can run it manually from the command line.

import argparse
import sys
import datetime
import subprocess
import logging
logging.basicConfig(filename='/Users/markberman/TextAnalytics/PythonCode/logging/HelloWorld.log',level=logging.DEBUG)

#total number of args passed
total = len(sys.argv)

logging.info("The total number of arguments passed to the script: %d" % total)

for i in xrange(0, total):
	logging.info("argument " + str(i) + " is " + sys.argv[i])

logging.info('Hello World')

Here is the error I receive when running logstash.

[2016-12-05T18:15:12,422][FATAL][logstash.runner          ] An unexpected error occurred! {:error=>#<NoMethodError: undefined method `terminal' for #<LogStash::Logging::Logger:0x662fad11>>, :backtrace=>["/Users/markberman/elastic5/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-exec-3.0.3/lib/logstash/outputs/exec.rb:58:in `receive'", "org/jruby/RubyIO.java:4475:in `popen3'", "/Users/markberman/elastic5/logstash/vendor/jruby/lib/ruby/1.9/open3.rb:74:in `popen3'", "/Users/markberman/elastic5/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-exec-3.0.3/lib/logstash/outputs/exec.rb:51:in `receive'", "/Users/markberman/elastic5/logstash/logstash-core/lib/logstash/outputs/base.rb:92:in `multi_receive'", "org/jruby/RubyArray.java:1613:in `each'", "/Users/markberman/elastic5/logstash/logstash-core/lib/logstash/outputs/base.rb:92:in `multi_receive'", "/Users/markberman/elastic5/logstash/logstash-core/lib/logstash/output_delegator_strategies/legacy.rb:19:in `multi_receive'", "/Users/markberman/elastic5/logstash/logstash-core/lib/logstash/output_delegator.rb:42:in `multi_receive'", "/Users/markberman/elastic5/logstash/logstash-core/lib/logstash/pipeline.rb:297:in `output_batch'", "org/jruby/RubyHash.java:1342:in `each'", "/Users/markberman/elastic5/logstash/logstash-core/lib/logstash/pipeline.rb:296:in `output_batch'", "/Users/markberman/elastic5/logstash/logstash-core/lib/logstash/pipeline.rb:252:in `worker_loop'", "/Users/markberman/elastic5/logstash/logstash-core/lib/logstash/pipeline.rb:225:in `start_workers'"]}
marks-MacBook-Pro:bin markberman$

**Comment by suyograo** [comment]
@markcberman can you try now? Sorry, I had missed a step to publish the changes. You'd have to install the plugin again, which will install v3.1.0

**Comment by markcberman** [comment]
I will reinstall and report back to you.

**Comment by markcberman** [comment]
same error.

**Comment by markcberman** [comment]
Is there any other debug information you need from me ?

**Comment by markcberman** [comment]
Here is a key piece of information.   

The logstash-output-exec plugin is successfully invoked ONE time and then it throws the error listed above and kills ./logstash.   (I got a single successful log file entry before the logstash-output-exec plugin blew up and killed ./logstash.


Here is the logstash.conf file I am using.

input {
     twitter {
        consumer_key => "XXXXXXXXX"
        consumer_secret => "XXXXXXXXX"
        keywords => ["Minnesota Vikings"]
        oauth_token => "XXXXXXXXX"
        oauth_token_secret => "XXXXXXXXXXX"
        #proxy_address => "XXXXXXXXXXXX"
        #proxy_port => 3128
        #use_proxy => true
        full_tweet => true
        ignore_retweets => true
    }
}
filter {
}
output {
   elasticsearch {
    hosts => ["127.0.0.1:9200"]
    action => "update"
    index => "twitter-*"
    document_type => "tweet"
    document_id => "%{id_str}-%{timestamp_ms}"
    #user => "xxx"
    #password => "xxxx"
    template => "/Users/markberman/elastic5/logstash/bin/twitter_template1.json"
    template_name => "twitter_cs410_project"
    doc_as_upsert => true
  }
  exec{
    command => 'python /Users/markberman/TextAnalytics/PythonCode/HelloWorld.py'
  }
}

**Comment by suyograo** [comment]
@markcberman will try in a bit. thanks for the config

**Comment by markcberman** [comment]
The setup described above works like a charm in Logstash version 2.4.1.  Seems like the issue I am reporting is isolated to the ELK 5.0 stack -- specifically Logstash 5.0.0.    Is there any time frame for fixing this issue, as I am not able to back version our Elastic indices to 2.x.

**Comment by markcberman** [comment]
@suyograo Any word when a fix is expected ?

**Comment by csobchuk** [comment]
Same problem. Would appreciate hearing when it is fixed for 5.2.
